### PR TITLE
Additional sensor support for Stulz wib

### PR DESCRIPTION
**Pull request by martinberg**
This is a new take on #16072 which I bodged and wasn't able to get in working order with my git and rebase fu.

The `unitFreecoolingValve` have gone from being a value of 0 or 1 in the old MIB to a value between 0 and 100 in the new.
Therefore there is a duplicate for this OID in the discovery file to not break anything.

Note: I'm unable to get snmpsim to work. Hopefully somebody can assist by creating the json test file.
I have redacted name of cooling units and IP adresses in the snmprec

```
[librenms@librenms-tst ~]$ ./scripts/save-test-data.php -o stulz
Waiting for snmpsim to initialize...
Failed to start snmpsim, make sure it is installed, working, and there are no bad snmprec files.
Run ./scripts/save-test-data.php --snmpsim to see the log output
```

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/definitions/discovery/stulz.yaml [hunk 1]
```diff
@@ -1,14 +1,19 @@
 mib: Stulz-WIB8000-MIB
 modules:
     sensors:
+        pre-cache:
+            data:
+                - 
+                    oid:
+                        -  unitsettingName
```
Comment (murrant) [inline includes/definitions/discovery/stulz.yaml:8]: This is unused.

**Comment by PipoCanaja** [comment]
Hi @martinberg 
Try to apply this PR in your setup :
https://github.com/librenms/librenms/pull/16267

This should enable debug and help you fix the issue.

@murrant : Should we merge it in order to help @martinberg  and others ? It doesn't seems to have any impact on working setups, and it only helps non-working ones.

**Comment by martinberg** [comment]
> Hi @martinberg Try to apply this PR in your setup to see the error in the SNMPSIM setup : #16267
> 
> This should enable debug and help you fix the issue.
> 
> @murrant : Should we merge it in order to help @martinberg and others ? It doesn't seems to have any impact on working setups, and it only helps non-working ones.

Thanks @PipoCanaja!
It made me aware of the issue however I'm not sure how to resolve it

```
[librenms@librenms-tst ~]$ ./scripts/save-test-data.php --os stulz -d
Setting up snmpsim virtual env in /opt/librenms/.python_venvs/snmpsim/
sh: line 0: exec: python: not found
Requirement already satisfied: snmpsim in ./.python_venvs/snmpsim/lib/python3.6/site-packages
Requirement already satisfied: pysnmp<5.0.0,>=4.4.3 in ./.python_venvs/snmpsim/lib/python3.6/site-packages (from snmpsim)
Requirement already satisfied: pyasn1>=0.2.3 in ./.python_venvs/snmpsim/lib/python3.6/site-packages (from pysnmp<5.0.0,>=4.4.3->snmpsim)
Requirement already satisfied: pycryptodomex in ./.python_venvs/snmpsim/lib/python3.6/site-packages (from pysnmp<5.0.0,>=4.4.3->snmpsim)
Requirement already satisfied: pysmi in ./.python_venvs/snmpsim/lib/python3.6/site-packages (from pysnmp<5.0.0,>=4.4.3->snmpsim)
Requirement already satisfied: ply in ./.python_venvs/snmpsim/lib/python3.6/site-packages (from pysmi->pysnmp<5.0.0,>=4.4.3->snmpsim)
You are using pip version 9.0.3, however version 24.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Waiting for snmpsim to initialize...
Failed to start snmpsim, make sure it is installed, working, and there are no bad snmprec files.
Run ./scripts/save-test-data.php --snmpsim to see the log output
```

It is quite expected.
I'm running RHEL 8.10 and have python2 and python3 installed.
However nothing links to python without a version number.

```
[librenms@librenms-tst ~]$ which python
/usr/bin/which: no python in (/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
[librenms@librenms-tst ~]$ which python2
/usr/bin/python2
[librenms@librenms-tst ~]$ which python3
/usr/bin/python3
```

The failed command seems to orignate from here however changing the line to `python3`
 or `python2` still resulted in a failed start of snmpsim.
 
 https://github.com/librenms/librenms/blob/40d0f68e3a958ff8c87853f6535773808516dee2/LibreNMS/Util/Snmpsim.php#L84

**Comment by martinberg** [comment]
> Hi @martinberg I am not at all a python guy :D and not a CentOS guy either ... Quick hack : create an alias `python` pointing to `python3`. If it works, you are good to go. Better hack, using `alternatives` mecanism. You can have a few details here : https://stackoverflow.com/questions/45542690/how-to-set-python3-5-2-as-default-python-version-on-centos

Thanks @PipoCanaja! :)
Python 3.6.8 was to old for some reason and snmpsim was unable to start.
Installed and linked Python 3.12.3 and snmpsim worked fine.

**Comment by martinberg** [comment]
@murrant Anything more that is need for this PR to be merged?

**Comment by laf** [comment]
Tests are failing because of the multiplier for the water flow. Is 16.6666666667 correct? If so, we'll need to update our tests to take that in to account.

**Comment by martinberg** [comment]
> Tests are failing because of the multiplier for the water flow. Is 16.6666666667 correct? If so, we'll need to update our tests to take that in to account.

The number is correct and is to convert m3/h to l/m. 
Rounding up to 17 is an option to make it an integer but this results in a value that is not quite true.

**Comment by martinberg** [comment]
Changed mutiplier to integer. Ready for review

**Comment by murrant** [comment]
What about setting multipler 50 and divisor 3?

**Review by laf — APPROVED** [review]
LGTM - Thanks for the contribution @martinberg

**Pull request merged**